lib/foreman/renderer/scope/macros/snippet_rendering.rb

```diff
@@ -9,29 +9,29 @@ def snippet_if_exists(name, options = {})
 
           # provide embedded snippets support as simple erb templates
           def snippets(file, options = {})
-            if ::Template.where(:name => file, :snippet => true).empty?
-              render :partial => "unattended/snippets/#{file}"
-            else
+            if @source.find_snippet(file)
               snippet(file.gsub(/^_/, ""), options)
+            else
+              render :partial => "unattended/snippets/#{file}"
             end
           end
 
           def snippet(name, options = {}, variables: {})
-            if (template = ::Template.where(:name => name, :snippet => true).first)
-              begin
-                snippet_variables = variables.merge(options[:variables] || {})
-                source = Foreman::Renderer.get_source(template: template, host: @host)
-                scope = Foreman::Renderer.get_scope(host: @host, variables: snippet_variables)
-                return Foreman::Renderer.render(source, scope)
-              rescue ::Foreman::Exception
-                raise
-              rescue StandardError => exc
-                e = ::Foreman::Exception.new(N_("The snippet '%{name}' threw an error: %{exc}"), { :name => name, :exc => exc })
-                e.set_backtrace exc.backtrace
-                raise e
-              end
-            else
+            template = @source.find_snippet(name)
+            unless template
               raise "The specified snippet '#{name}' does not exist, or is not a snippet." unless options[:silent]
+              return
+            end
+
+            begin
+              snippet_variables = variables.merge(options[:variables] || {})
+              template.render(host: host, variables: snippet_variables, mode: mode)
+            rescue ::Foreman::Exception
+              raise
```

Comment:
why we rescue from exception here? also by just raising new runtime error, we lose the original backtrace, should we perhaps log it?

Reply:
I don't know and am wondering the same thing. It's from the original PR.

I was wondering if maybe this is so that the next rescue block does not catch the exception. And a `Foreman::Exception` is not wrapped in another `Foreman::Exception`.

But we can add logging, sure.